gateway/src/lib.rs

```diff
@@ -820,146 +821,21 @@ impl Actor {
     /// of the total number of validators, the messages will be applied.
     ///
     /// Each cron checkpoint will be checked against each other using blake hashing.
-    fn submit_cron(rt: &mut impl Runtime, params: CronCheckpoint) -> Result<RawBytes, ActorError> {
+    fn submit_cron(
+        rt: &mut impl Runtime,
+        checkpoint: CronCheckpoint,
+    ) -> Result<RawBytes, ActorError> {
         // submit cron can only be performed by signable addresses
         rt.validate_immediate_caller_type(CALLER_TYPES_SIGNABLE.iter())?;
 
-        let params_epoch = params.epoch;
+        Self::execute_next_cron_epoch(rt)?;
```

Comment:
Take this down and perform only one execution per submission. Add a note of why we do this.